compiler/p/codegen/OMRMemoryReference.cpp

```diff
@@ -1340,7 +1341,15 @@ TR::Instruction *OMR::Power::MemoryReference::expandInstruction(TR::Instruction
 
          if (displacement < LOWER_IMMED || displacement > UPPER_IMMED)
             {
-            prevInstruction = generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addis, node, base, base, cg->hiValue(displacement), prevInstruction);
+            if (0x00008000 == cg->hiValue(displacement))
+               {
+               generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addis, node, base, base, 0x7FFF);
+               generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addis, node, base, base, 0x1);
```

Comment:
These instructions need to be generated using `prevInstruction` as the cursor argument and need to be assigned to `prevInstruction` as they are generated.